src/main/kotlin/com/hj/leetcode/kotlin/problem1372/Solution.kt

```diff
@@ -0,0 +1,31 @@
+package com.hj.leetcode.kotlin.problem1372
+
+import com.hj.leetcode.kotlin.common.model.TreeNode
+
+/**
+ * LeetCode page: [1372. Longest ZigZag Path in a Binary Tree](https://leetcode.com/problems/longest-zigzag-path-in-a-binary-tree/);
+ */
+class Solution {
+    /* Complexity:
+     * Time O(N) and Space O(H) where N and H are the number of nodes and height of root
+     */
+    fun longestZigZag(root: TreeNode?): Int {
+        var longestZigZag = 0
+        dfsZigZag(root, 0, 0) { maxLengthToNode ->
+            if (maxLengthToNode > longestZigZag) longestZigZag = maxLengthToNode
+        }
+        return longestZigZag
+    }
+
+    private fun dfsZigZag(
+        root: TreeNode?,
+        lengthFromLeftEdge: Int,
+        lengthFromRightEdge: Int,
```

Comment:
`lengthFromLeftEdge` and `lengthFromRightEdge` are not clear and should find a better name

Reply:
Struggled but cannot come out with good names. Comments have been added to made it clearer.